plutus-ir/src/Language/PlutusIR/Compiler/Types.hs

```diff
@@ -77,3 +80,10 @@ type PIRType a = PIR.Type PIR.TyName (Provenance a)
 type Compiling m e a = (Monad m, MonadReader (CompilationCtx a) m, AsError e (Provenance a), MonadError e m, MonadQuote m)
 
 type TermDef tyname name a = PLC.Def (PLC.VarDecl tyname name a) (PIR.Term tyname name a)
+
+-- | We generate some shared definitions compilation, this datatype
+-- defines the "keys" for those definitions.
+data SharedName = FixpointCombinator Integer deriving (Show, Eq, Ord)
```

Comment:
This is terribly ad hoc.

Reply:
Not really. This is just an enumeration of the things that we generate repeatedly in the PIR compiler. I'll probably add `Unit` at some point, and that may well be it.

Otherwise I could put `Text` here, but that seems worse than a datatype when we know the cases.